File: src/backend/GetToKnowGame/Repositories/QuestionRepository.cs
using GetToKnowGame.Models;
using MongoDB.Driver;

namespace GetToKnowGame.Repositories
{
    public class QuestionRepository : BaseRepository<Question>, IQuestionRepository
    {
        public QuestionRepository(IMongoDatabase database) : base(database, "questions")
        {
        }

        public async Task<IEnumerable<Question>> GetBySectionAsync(string section)
        {
            var filter = Builders<Question>.Filter.Eq(q => q.Section, section);
            return await _collection.Find(filter).ToListAsync();
        }

        public async Task<IEnumerable<string>> GetSectionsAsync()
        {
            var sections = await _collection.Distinct<string>("section", FilterDefinition<Question>.Empty).ToListAsync();
            return sections;
        }
    }
}

File: src/backend/get-to-know-game-go/repositories/question_repository.go
package repositories

import (
	"get-to-know-game-go/models"

	"go.mongodb.org/mongo-driver/mongo"
)

// QuestionRepositoryImpl implements QuestionRepository
type QuestionRepositoryImpl struct {
	*BaseRepository[models.Question]
}

// NewQuestionRepository creates a new question repository
func NewQuestionRepository(collection *mongo.Collection) QuestionRepository {
	return &QuestionRepositoryImpl{
		BaseRepository: NewBaseRepository[models.Question](collection),
	}
}
